components/director/pkg/certloader/loader.go

```diff
@@ -61,23 +89,34 @@ func StartCertLoader(ctx context.Context, externalClientCertSecret string) (Cach
 	if err != nil {
 		return nil, err
 	}
-	certCache := NewCertificateCache()
-	certLoader := NewCertificatesLoader(certCache, k8sClientSet.CoreV1().Secrets(parsedCertSecret.Namespace), parsedCertSecret.Name, time.Second)
+	certLoader := NewCertificateLoader(certLoaderConfig, certCache, k8sClientSet.CoreV1().Secrets(parsedCertSecret.Namespace), parsedCertSecret.Name, time.Second)
 	go certLoader.Run(ctx)
 
 	return certCache, nil
 }
 
+// Run populates the certificate cache on local setup with provided certificate and key
```

Comment:
its normally not ok to have local dev code in the prod code base, but i havent looked at the prev PR. Is this the only way you can think of for this to be achieved ?

Reply:
One idea - you could try to install https://github.com/k3s-io/k3s#quick-start---install-script as part of the local run.sh script and then point the kubeconfig to point to it. And in the run.sh also always create a certificate in the relevant secret

Reply:
That's the thing that I came up with. Most probably there is another way to do it but in the productive setup(and local installation) we use k8s cluster with watch mechanism. The "problem" with the director is that we run it standalone and don't have k8s cluster. I can check what are the alternatives

Reply:
I've missed the previous comment. About the disabled certloader locally, that was the initial idea to return empty cache. And currently in the PR that's the default behaviour but I introduce a functionality where we can add custom certificate in the director environment(run.sh script) so we can populate the cache with it. The motivation behind this was if we want to run an ord app locally + director(run.sh) which is configured with client cert issued by CA used in the ord app. I'm not sure if we have such case though.